src/commandpalette/plugin.ts

```diff
@@ -12,10 +17,55 @@ import {
 } from '../application';
 
 import {
-  ICommandPalette
+  ICommandPalette, IPaletteItem
 } from './';
 
 
+/**
+ * A thin wrapper around the `CommandPalette` class to conform with the
+ * JupyterLab interface for the application-wide command palette.
+ */
+class Palette implements ICommandPalette {
+  /**
+   * Create a palette instance.
+   */
+  constructor(palette: CommandPalette) {
+    this._palette = palette;
+  }
+
+  /**
+   * The placeholder text of the command palette's search input.
+   */
+  set placeholder(placeholder: string) {
+    this._palette.inputNode.placeholder = placeholder;
+  }
+  get placeholder(): string {
+    return this._palette.inputNode.placeholder;
+  }
+
+  /**
+   * Activate the command palette for user input.
+   */
+  activate(): void {
+    this._palette.inputNode.focus();
```

Comment:
Should call `this._palette.activate()`: https://github.com/phosphorjs/phosphor/blob/master/src/ui/commandpalette.ts#L339